content/actions/use-cases-and-examples/building-and-testing/building-and-testing-rust.md

```diff
@@ -0,0 +1,141 @@
+{% endif %}
+
+## Specifying a Rust version
+
+At the time of writing, the default rust compiler version is 1.83.0 rustup is available and can be used to install additional toolchains.
+
+```yaml copy
+      - name: Temporarily modify the rust toolchain version
+        run: rustup override set nightly
+      - name: Ouput rust version for educational purposes
+        run: rustup --version
+```
+
+### Caching dependencies
+
+You can cache and restore dependencies using the following example below. Note that you will need to have Cargo.lock in your repository to cache dependencies.
+
+```yaml copy
+      - name: ⚡ Cache
+      - uses: {% data reusables.actions.action-cache %}
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: {% raw %}${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}{% endraw %}
+```
+
+If you have a custom requirement or need finer controls for caching, you can take a look at the [`cache` action](https://github.com/marketplace/actions/cache). For more information, see [AUTOTITLE](/actions/using-workflows/caching-dependencies-to-speed-up-workflows).
+
+## Building and testing your code
+
+You can use the same commands that you use locally to build and test your code. This example workflow demonstrates how to use `cargo build` and `cargo test` in a job:
+
+```yaml copy
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        BUILD_TARGET: [release] # refers to a cargo profile
+    outputs:
+      release_built: ${{ steps.set-output.outputs.release_built }}
+    steps:
+      - uses: {% data reusables.actions.action-checkout %}
+      - name: Build binaries in "${{ matrix.BUILD_TARGET }}" mode
+        run: cargo build --profile ${{ matrix.BUILD_TARGET }}
+      - name: Run tests in "${{ matrix.BUILD_TARGET }}" mode
+        run: cargo test --profile ${{ matrix.BUILD_TARGET }}
+```
+
+Note that the `release` keyword used above, corresponds to a cargo profile. You can use any [profile](https://doc.rust-lang.org/cargo/reference/profiles.html) you have defined in your `Cargo.toml` file.
+
+## Upload artifacts
+
+In case publishing artifacts is needed, but not to crates.io, the following example demonstrates how to upload artifacts to the workflow run:
+
+```yaml copy
+      - name: Upload hello app
+        uses: {% data reusables.actions.action-upload-artifact %}
+        with:
+          name: cndk8-hello
+          path: target/${{ matrix.BUILD_TARGET }}/cndk8
+```
+
+And to use them on a different job, i.e publishing:
+
+```yaml copy
+      - name: Download hello app
+        uses: {% data reusables.actions.action-download-artifact %}
+        with:
+          name: cndk8-hello
+          path: ./cndk8-hello
+      - name: Publish hello app to GitHub Packages
+        run: |
+          curl -u "${{ github.actor }}:${{ secrets.GH_TOKEN }}" \
```

Comment:
Should this be `GITHUB_TOKEN` here, rather than `GH_TOKEN`?
```suggestion
          curl -u "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}" \
```